src/components/assignment-panel/assignment-panel.tsx

```diff
@@ -1,18 +1,27 @@
-import React, { Fragment, ReactNode, useMemo } from 'react'
-import { ArrowBackSharp } from '@material-ui/icons'
+import React, { Fragment, ReactNode, useCallback, useEffect, useMemo, useState } from 'react'
+import { CircularProgress } from '@material-ui/core'
+import { Popover, Tooltip } from 'antd'
+import { ArrowBackSharp, SupervisorAccountOutlined, SyncOutlined } from '@material-ui/icons'
 import {
     panelWrapperClass,
-    panelHeaderClass
+    panelHeaderClass,
+    studentManagementPopoverOverlayClass
 } from './style'
 import { AssignmentContent } from './assignment-content'
-import { useAssignment, useCommands } from '../../contexts'
+import { useAssignment, useCommands, useSettings, useSnackbar } from '../../contexts'
+import { syncToLMS } from '../../api'
 
 interface IAssignmentPanelProps {
 }
 
 export const AssignmentPanel = ({}: IAssignmentPanelProps) => {
     const commands = useCommands()!
-    const { course, assignment } = useAssignment()!
+    const snackbar = useSnackbar()!
+    const { repoRoot } = useSettings()!
+    const { course, students, assignment } = useAssignment()!
+    
+    const [syncLoading, setSyncLoading] = useState<boolean>(false)
+
     const headerName = useMemo<ReactNode>(() => {
         const headerFragments = []
         // if (assignment) headerFragments.push(assignment.name)
```

Comment:
did you mean to delete this?

Reply:
No, why do you think that?

Reply:
It's commented out... 

We generally shouldn't keep around commented code without a comment at the top explaining why it's being kept around